kroxylicious-api/src/main/java/io/kroxylicious/proxy/filter/KrpcFilter.java

```diff
@@ -11,4 +11,11 @@
  */
 public /* sealed */ interface KrpcFilter {
 
+    /**
+     * Called when the Filter is removed, for instance when the client connection is closed
+     */
+    default void onFilterRemoved() {
```

Comment:
I think the implementation of `onFIlterRemove` should have a way to access things like the `io.kroxylicious.proxy.internal.DefaultFilterContext#channelDescriptor`